### PR TITLE
Added partitioning capability in SpannerToGCS template

### PR DESCRIPTION
**Pull request by vanshaj-bhatia**

### CODE REVIEW

**Review by shashank-google — CHANGES_REQUESTED** [review]
Does not looks right. Only adding partitioning column will not work.
did you executed the template and observed that multiple files are outpoted against single table ?
Refer to JdbcToGCS template blogpost to learn about partitioning and how it works.

**Comment by vanshaj-bhatia** [comment]
I misunderstood the requirement, my bad. Added the SQL partitioning capabilities now and also included the Jakarta validator at SpannerToGCSConfig.

File: java/src/main/java/com/google/cloud/dataproc/templates/databases/README.md [hunk 1]
```diff
@@ -14,10 +14,18 @@ bin/start.sh \
 --templateProperty spanner.gcs.input.database.id=<database-id> \
 --templateProperty spanner.gcs.input.table.id=<table-id> \
 --templateProperty spanner.gcs.output.gcs.path=<gcs-path> \
---templateProperty spanner.gcs.output.gcs.saveMode=<Append|Overwrite|ErrorIfExists|Ignore>
---templateProperty spanner.gcs.output.gcs.format=<avro|csv|parquet|json|orc>
+--templateProperty spanner.gcs.output.gcs.saveMode=<Append|Overwrite|ErrorIfExists|Ignore> \
+--templateProperty spanner.gcs.output.gcs.format=<avro|csv|parquet|json|orc> \
+--templateProperty spanner.gcs.output.gcs.partitionColumn=<optional-partition-column> \
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/databases/README.md:19]: we don't want to have `output.gcs.partitionColumn`

File: java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java [hunk 2]
```diff
@@ -27,6 +27,11 @@ public interface TemplateConstants {
   String SPANNER_GCS_OUTPUT_GCS_PATH = "spanner.gcs.output.gcs.path";
   String SPANNER_GCS_OUTPUT_GCS_SAVEMODE = "spanner.gcs.output.gcs.saveMode";
   String SPANNER_GCS_OUTPUT_FORMAT = "spanner.gcs.output.gcs.format";
+  String SPANNER_GCS_OUTPUT_PARTITION_COLUMN = "spanner.gcs.output.gcs.partitionColumn";
```
Comment (shashank-google) [inline java/src/main/java/com/google/cloud/dataproc/templates/util/TemplateConstants.java:30]: remove code for output.gcs.partitionColumn

File: java/src/main/resources/template.properties [hunk 3]
```diff
@@ -135,12 +135,18 @@ gcs.bigquery.output.table=
...
-spanner.gcs.output.gcs.path=<gcs-path>
-spanner.gcs.output.gcs.saveMode=<Append|Overwrite|ErrorIfExists|Ignore>
-spanner.gcs.output.gcs.format=<avro|csv|parquet|json|orc>
+spanner.gcs.input.spanner.id=
+spanner.gcs.input.database.id=
+spanner.gcs.input.table.id=
+spanner.gcs.output.gcs.path=
+spanner.gcs.output.gcs.saveMode=
+spanner.gcs.output.gcs.format=
+spanner.gcs.output.gcs.partitionColumn=
```
Comment (shashank-google) [inline java/src/main/resources/template.properties:144]: remove code for output.gcs.partitionColumn

**Review by shashank-google — APPROVED** [review]
I have approved it.
But please remove any code regarding `output.gcs.partitionColumn` before checking in.
Also please test run the template before merging.

File: java/src/main/java/com/google/cloud/dataproc/templates/databases/README.md (hunk 1 above)
Comment (vanshaj-bhatia) [reply java/src/main/java/com/google/cloud/dataproc/templates/databases/README.md]: Removed.

File: java/src/main/java/com/google/cloud/dataproc/templates/databases/SpannerToGCS.java [hunk 4]
```diff
@@ -33,59 +37,59 @@ public class SpannerToGCS implements BaseTemplate {
...
+    if (StringUtils.isNotBlank(config.getTempTable())
+        && StringUtils.isNotBlank(config.getTempQuery())) {
+      jdbcDF.createOrReplaceGlobalTempView(config.getTempTable());
+      jdbcDF = spark.sql(config.getTempQuery());
+    }
+
+    DataFrameWriter<Row> writer =
+        jdbcDF.write().format(config.getGcsOutputFormat()).mode(config.getGcsWriteMode());
+
+    if (StringUtils.isNotBlank(config.getGcsPartitionColumn())) {
```
[the remark this answers is not in the document]
Comment (vanshaj-bhatia) [reply java/src/main/java/com/google/cloud/dataproc/templates/databases/SpannerToGCS.java]: Removed.

**Pull request merged**